### PR TITLE
Support for "Enable Searching" property of groups

### PR DESCRIPTION
**Pull request by infeeeee**
Hello

Added support for reading and setting the `EnableSearching` parameter. Other keepass clients use this to set that search should find results in that group. 

On KepassXC and Keepass the trash group has this option enabled by default, I added the same behavior as well.

I didn't modify the `find_`... functions. Should I add support there as well, or only clients should implement this search filtering?

I added some tests for this new thing, and added some generic group attribute tests based on the similar entry tests. Should I add more, or is this enough?

Edit: TODO:

- [x] Rename to `searching_enabled`
- [ ] Add support in the `find_`... functions
- [ ] Add tests for the `find_`... functions
- [ ] Update readme

### CODE REVIEW

**Comment by Evidlo** [comment]
Thanks for the contribution.

> I didn't modify the find_... functions. Should I add support there as well, or only clients should implement this search filtering?

That would definitely be desirable.  I think you only need to modify `xpath.py` to add this.  See `autotype_enabled` as an example.  Also add this to this `find_groups` section in the README.

I think `searching_enabled` would be a better name for this property so that it matches existing vocabulary like `autotype_enabled`.

File: pykeepass/pykeepass.py [hunk 1]
```diff
@@ -434,13 +434,16 @@ def find_groups(self, recursive=True, path=None, group=None, **kwargs):
         return res
 
     # creates a new group and all parent groups, if necessary
-    def add_group(self, destination_group, group_name, icon=None, notes=None):
+    def add_group(self, destination_group, group_name, icon=None, 
+			notes=None, enable_searching=None):
```
Comment (Evidlo) [inline pykeepass/pykeepass.py:438]: Please use spaces instead of tabs.

Comment (infeeeee) [reply pykeepass/pykeepass.py]: Fiexd

**Comment by Evidlo** [comment]
If you need any more direction on what to do, let me know.

**Comment by infeeeee** [comment]
I didn't have time nowadays to work on this, but I still want to finish it sometimes in the future. This xpath is a bit confusing, I used lxml before on another project, but never used this part.

If you want to finish it, I won't be mad :)

**Comment by infeeeee** [comment]
I stopped using keepass and switched to a different password manager, so I won't finish this PR, sorry.

**Pull request closed without merging**